### ISSUE TITLE
ArgoCD Manifest error

### ISSUE THREAD

**Issue by brimdor** [khuedoan/homelab#160]
Similar issue to: https://github.com/khuedoan/homelab/issues/102
When running "make" for the first time, when it gets to the "system" portion, it fails.

system\bootstrap.yml
- name: Apply ArgoCD manifests
      kubernetes.core.k8s:
        resource_definition: "{{ argocd_manifests.stdout }}"
        apply: true
        server_side_apply:
          field_manager: argocd-controller

That task fails and gives the following error:
TASK [Apply ArgoCD manifests] ******************************************************************************************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: ansible_collections.kubernetes.core.plugins.module_utils.k8s.exceptions.CoreException: Failed to load resource definition: 'str' object has no attribute 'setdefault'
fatal: [localhost]: FAILED! => {"changed": false, "msg": "Failed to load resource definition: 'str' object has no attribute 'setdefault'"}

To reproduce this error, start over from scratch, to replicate a first time build. You can do this using my reset script I put in a pull request along with other quality of life change suggestions. (https://github.com/khuedoan/homelab/pull/159)

Then run the usual process to build.
1. make tools
2. make configure (if needed)
3. make

Once it gets to the step described above, it will fail.

**Comment by fullstackjam** [comment]
You can retry run `make`, it will work

**Comment by brimdor** [comment]
Yes but that defeats the purpose of a single command build. You see this
being performed in the video in the readme. That is not possible if the
process has a break and then you have to run it again.

**Comment by fullstackjam** [comment]
- name: argo-cd
    version: 5.55.0
works great

**Comment by fullstackjam** [comment]
interesting thing is when i use [kubepary](https://github.com/kubernetes-sigs/kubespray) to install kubernetes: v1.30.3, `make -C system` works well, so i think question may be on k3s version

**Comment by isvicy** [comment]
> interesting thing is when i use [kubepary](https://github.com/kubernetes-sigs/kubespray) to install kubernetes: v1.30.3, `make -C system` works well, so i think question may be on k3s version

@fullstackjam  hi, do you maintain your own homelab bootstrap of using kubepary to bootstrap the metal? it could be very awesome if so.
